catalog/dags/data_refresh/alter_data.py

```diff
@@ -0,0 +1,209 @@
+
+    if update_required:
+        return tag_output
+    return None
+
+
+@task
+def get_alter_batches(
+    id_bounds: tuple[int, int], batch_size: int | None
+) -> list[tuple[int, int]]:
+    start, stop = id_bounds
+    batch_size = batch_size or DEFAULT_BATCH_SIZE
+    return [(x, x + batch_size - 1) for x in range(start, stop, batch_size)]
+
+
+@task
+def alter_data_batch(
+    batch: tuple[int, int],
+    temp_table: str,
+    postgres_conn_id: str,
+    task: AbstractOperator = None,
+    timeout: float = None,
+) -> int:
+    logger.info(f"Starting data altering on batch: {batch}")
+    batch_start, batch_end = batch
+    postgres = PostgresHook(
+        postgres_conn_id=postgres_conn_id,
+        default_statement_timeout=(
+            timeout if timeout else PostgresHook.get_execution_timeout(task)
+        ),
+    )
+    rows = postgres.run(
+        QUERY_SELECTION.format(
+            temp_table=temp_table, batch_start=batch_start, batch_end=batch_end
+        ),
+        handler=fetch_all_handler,
+    )
+    worker_conn = postgres.get_conn()
+    logger.info("Data altering worker connected to database")
+    with worker_conn.cursor() as write_cur:
+        logger.info(f"Filtering {len(rows)} rows")
+
+        altered_count = 0
+        for row in rows:
+            _id, identifier, tags = row
+            tags_fragment = generate_tag_updates(tags)
+            if not tags_fragment:
+                continue
+            altered_count += 1
+            logger.debug(
+                f"Updated tags for {identifier}\n\t"
+                f"from '{tags}' \n\tto '{tags_fragment}'"
+            )
+            update_query = QUERY_UPDATE.format(
+                temp_table=temp_table, tags_fragment=Json(tags_fragment), _id=_id
+            )
+            logger.debug(f"Executing update query: \n\t{update_query}")
+            write_cur.execute(update_query)
+    logger.info("Worker committing changes...")
+    worker_conn.commit()
+    worker_conn.close()
+    return altered_count
+
+
+@task
+def report(counts: list[int]):
+    total_count = sum(counts)
+    logger.info(f"Filtered tags for a total of {total_count} records")
+    return total_count
+
+
+@task_group(group_id="alter_table_data")
+def alter_table_data(
+    environment: Environment,
+    data_refresh_config: DataRefreshConfig,
+):
+    """Perform data altering across a number of tasks."""
+    postgres_conn_id = POSTGRES_API_CONN_IDS.get(environment)
+    temp_table = data_refresh_config.table_mappings[0].temp_table_name
+
+    estimated_record_count = PGExecuteQueryOperator(
+        task_id="get_estimated_record_count",
+        conn_id=postgres_conn_id,
+        sql=QUERY_ROW_ESTIMATE.format(temp_table=temp_table),
+        handler=fetch_one_handler,
+        return_last=True,
+    )
+
+    batches = get_alter_batches(
+        estimated_record_count.output,
+        batch_size="{{ var.value.get('DATA_REFRESH_ALTER_BATCH_SIZE', none) }}",
```

Comment:
Shame it's tricky to default this, but I think the way you've done it is the best option 🤔

Reply:
Yea...I agree 😞 This seemed like the best way to make sure that it was 1) only filled in by Airflow at runtime and 2) easiest for testing different batch sizes.